pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -252,7 +261,7 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 				MountPath: constants.DefaultModelLocalMountPath,
 				// only path to volume's root ("") or folder is supported
 				SubPath:  pvcPath,
-				ReadOnly: true,
+				ReadOnly: isvcReadonlyStringFlag,
```

Comment:
how about this [ReadOnly](https://github.com/kserve/kserve/blob/b3019774251f8c64d3c1863c6564c2ef7b6592b5/pkg/webhook/admission/pod/storage_initializer_injector.go#L310) parameter, need to it as well?

Reply:
Hi @lizzzcai this one refers to the storage Initializer, I guess it is only reads the model directory.

Reply:
noob question: Could you clarify your use case? From https://github.com/kserve/kserve/issues/3687,  Do you download in the predictor container? 
In our use case, we would like to use the storage initializer to download to the pvc. So we'd like to make this [ReadOnly](https://github.com/kserve/kserve/blob/b3019774251f8c64d3c1863c6564c2ef7b6592b5/pkg/webhook/admission/pod/storage_initializer_injector.go#L310) parameter configurable, and ideally keep these two ReadOnly (L264 and L373) false.

Reply:
noob question: Could you clarify your use case? From #3687, Do you download in the predictor container?
In our use case, we would like to download to pvc in the storage initializer, which means we probably only want this [ReadOnly](https://github.com/kserve/kserve/blob/b3019774251f8c64d3c1863c6564c2ef7b6592b5/pkg/webhook/admission/pod/storage_initializer_injector.go#L310) parameter to be `false`. Is it a good idea to differentiate these two use cases?

Reply:
Hi @spolti I am not sure if there is a scenario that the user want to modify the existing file in the PVC, and whether it will work with the `ReadOnly: true` [here](https://github.com/kserve/kserve/blob/b3019774251f8c64d3c1863c6564c2ef7b6592b5/pkg/webhook/admission/pod/storage_initializer_injector.go#L310) and the symlink file under the storage initializer approach of mounting pvc.
With the changes [here](https://github.com/kserve/kserve/blob/b3019774251f8c64d3c1863c6564c2ef7b6592b5/pkg/webhook/admission/pod/storage_initializer_injector.go#L264), the above scenario should work with the `directPvcVolumeMount` which is the default way of mounting PVC in KServe now.

Hi @greenmoon55, if my understanding correctly, your scenario is downloading a model from cloud storage via storage initializer to a PVC rather than emptyDir, like what I described [here](https://github.com/kserve/kserve/issues/3687#issuecomment-2109717856). If yes, I think it is a different use case from this PR.

Reply:
To satisfy all use cases we might need to set it for storage-initializer as well.

@hdefazio can you please handle it?

Reply:
@spolti @lizzzcai I have updated this to include the storage-initializer if you'd like to double check